File: GarageGo-BE/Models/HoaDon.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GarageGo_BE.Models
{
    [Table("HOADON")]
    public class HoaDon
    {
        [Key]
        public int MaHD { get; set; }

        [Required]
        public int MaKH { get; set; }

        public int? MaXe { get; set; } // Make optional for product orders

        public int? UserId { get; set; } // Make optional

        public DateTime NgayLap { get; set; } = DateTime.Now;

        [Column(TypeName = "decimal(18,2)")]
        public decimal TongTien { get; set; } = 0;

        [StringLength(50)]
        public string? HinhThucTT { get; set; }

        [StringLength(30)]
        public string? TrangThai { get; set; }

        [NotMapped]
        [StringLength(50)]
        public string? Username => User?.Username; // Username của user đã mua (computed from User)

        [ForeignKey("MaKH")]
        public KhachHang? KhachHang { get; set; }

        [ForeignKey("MaXe")]
        public Xe? Xe { get; set; }

        [ForeignKey("UserId")]
        public User? User { get; set; }

        // ✅ Sửa ICollection thành List
        public List<ChiTietHDDV>? ChiTietDichVus { get; set; }
        public List<ChiTietHDSP>? ChiTietSanPhams { get; set; }
    }
}

File: GarageGo-BEJava/src/main/java/com/garagego/model/HoaDon.java
package com.garagego.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "HOADON")
public class HoaDon {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "MaHD")
    private Integer maHD;

    @Column(name = "MaKH", nullable = false)
    private Integer maKH;

    @Column(name = "MaXe")
    private Integer maXe;

    @Column(name = "UserId")
    private Integer userId;

    @Column(name = "NgayLap")
    private LocalDateTime ngayLap = LocalDateTime.now();

    @Column(name = "TongTien", precision = 18, scale = 2)
    private BigDecimal tongTien = BigDecimal.ZERO;

    @Column(name = "HinhThucTT", length = 50)
    private String hinhThucTT;

    @Column(name = "TrangThai", length = 30)
    private String trangThai = "Chờ xác nhận";

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "MaKH", insertable = false, updatable = false)
    private KhachHang khachHang;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "MaXe", insertable = false, updatable = false)
    private Xe xe;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "UserId", insertable = false, updatable = false)
    private User user;

    @OneToMany(mappedBy = "hoaDon", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<ChiTietHDDV> chiTietDichVus;

    @OneToMany(mappedBy = "hoaDon", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<ChiTietHDSP> chiTietSanPhams;
}
